projects/gloo/pkg/plugins/listener/plugin.go

```diff
@@ -39,6 +41,18 @@ func (p *plugin) ProcessListener(_ plugins.Params, in *v1.Listener, out *envoy_c
 	if in.GetOptions().GetSocketOptions() != nil {
 		out.SocketOptions = translateSocketOptions(in.GetOptions().GetSocketOptions())
 	}
+	if connectionBalanceConfig := in.GetOptions().GetConnectionBalanceConfig(); connectionBalanceConfig != nil {
+
+		if connectionBalanceConfig.GetExactBalance() != nil {
+			out.ConnectionBalanceConfig = &envoy_config_listener_v3.Listener_ConnectionBalanceConfig{
+				BalanceType: &envoy_config_listener_v3.Listener_ConnectionBalanceConfig_ExactBalance_{
+					ExactBalance: &envoy_config_listener_v3.Listener_ConnectionBalanceConfig_ExactBalance{},
+				},
+			}
+		} else if connectionBalanceConfig.GetBalanceType() == nil {
+			return errors.New("connection balancer does not specify balancer type")
+		}
+	}
```

Comment:
I think if you removed the `oneof` above this could be simplified quite a lot
```suggestion
	if in.GetOptions().GetConnectionBalanceConfig().GetExactBalance() != nil {
		out.ConnectionBalanceConfig = &envoy_config_listener_v3.Listener_ConnectionBalanceConfig{
			BalanceType: &envoy_config_listener_v3.Listener_ConnectionBalanceConfig_ExactBalance_{
				ExactBalance: &envoy_config_listener_v3.Listener_ConnectionBalanceConfig_ExactBalance{},
			},
		}
	}
```